doc/api/customization.md

```diff
@@ -0,0 +1,60 @@
+# Customization
+
+> Stability: 1.0 - Early development
+
+Node.js has multiple APIs for customizing behavior and can easily be invoked via
+[`--import`][]. The package.json of such a plugin would look something like
+this:
+
+```json
+{
+  "name": "example-nodejs-plugin",
+  "keywords": [
+    "nodejs-plugin"
+  ],
+  "exports": {
+    "register": "./registration.mjs"
+  }
+}
+```
+
+Setting the keyword `nodejs-plugin` is important for users to find the
+package (which may be automated). It should also contain the official name(s)
+for the support it provides; a plugin extending Node.js support for TypeScript
+should contain `typescript` in its keywords; a customization package that does
+not extend Node.js support for TypeScript should not contain `typescript` (such
+as when the library is itself merely written in TypeScript). Incorrect keywords
+will cause a package to be listed for things it does not do, aggravating users.
+
+## Setting up a plugin
+
+These steps configure Node.js with a customization plugin:
+
+1. Install the plugin via your preferred package manager. For finding the
+   plugin, package managers provide often a CLI utility, such as [`npm
+   search`][], as well as a website.
+2. Once installed, in order to get Node.js to use it, you have 2 options:
```

Comment:
A third option for Yarn users is to add the `NODE_OPTIONS` variable to `.env.yarn`, which gets [automatically injected into all `yarn node` calls](https://yarnpkg.com/configuration/yarnrc#injectEnvironmentFiles).

Reply:
Hmm, this may be more convenient, but if it works for only a minority of uses, I think it would be better for node to have 1 implementation that works for all.